src/test/java/picard/sam/SplitSamBySizeTest.java

```diff
@@ -71,8 +68,13 @@ public void testOkFile() throws IOException {
         final SAMRecordIterator inputIter = input.iterator();
         final SamReader reader1 = SamReaderFactory.makeDefault().open(out1);
         final SamReader reader2 = SamReaderFactory.makeDefault().open(out2);
+        final SAMFileHeader inputHeader = input.getFileHeader();
+        final SAMFileHeader reader1Header = reader1.getFileHeader();
 
-        Assert.assertEquals(input.getFileHeader(), reader1.getFileHeader());
+        // The output BAM versions might not match the input header version if the input is outdated.
+        inputHeader.setAttribute(SAMFileHeader.VERSION_TAG, reader1Header.getVersion());
```

Comment:
When the version changes can you emit a warning?

Reply:
done